src/zarr/core/group.py

```diff
@@ -1407,6 +1395,42 @@ async def _members(
         ):
             yield member
 
+    # TODO: find a better name for this. create_tree could work.
+    # TODO: include an example in the docstring
+    async def create_hierarchy(
+        self,
+        nodes: dict[str, ArrayV2Metadata | ArrayV3Metadata | GroupMetadata],
+        overwrite: bool,
+    ) -> AsyncIterator[AsyncGroup | AsyncArray[ArrayV2Metadata] | AsyncArray[ArrayV3Metadata]]:
+        """
+        Create a hierarchy of arrays or groups rooted at this group.
+
+        This method takes a dictionary where the keys are the names of the arrays or groups
+        to create and the values are the metadata or objects representing the arrays or groups.
+
+        The method returns an asynchronous iterator over the created nodes.
```

Comment:
A note about when to use this would be great

Reply:
good point, would something like "use this method to create an entire tree of sub-groups and / or sub-arrays efficiently." suffice?